src/System Application/App/Permission Sets/src/PermissionSetRelationImpl.Codeunit.al

```diff
@@ -364,6 +398,22 @@ codeunit 9856 "Permission Set Relation Impl."
         end;
     end;
 
+    local procedure GetPermissionSetName(AppId: Guid; RoleId: Code[30]; Scope: Option System,Tenant): Text[30];
+    var
+        TenantPermissionSet: Record "Tenant Permission Set";
+        MetaDataPermissionSet: Record "Metadata Permission Set";
+    begin
+        if Scope = Scope::Tenant then
+            if TenantPermissionSet.Get(AppId, RoleId) then
+                exit(TenantPermissionSet.Name);
+
+        if Scope = Scope::System then
+            if MetaDataPermissionSet.Get(AppId, RoleId) then
+                exit(MetaDataPermissionSet.Name);
+
+        exit('');
```

Comment:
Should we return something like "Unavailable"? or something that indicates this permission set does not exist to the user...

Reply:
Maybe this should be handled on the page actually.

Reply:
By definition it should always be in one of the two tables. However, in the unknown scenario where we can't get it in either, I believe an empty string is a decent fallback in that scenario.